bfcfg

```diff
@@ -484,13 +504,13 @@ mfg_cfg()
   local bsb_sn_sysfs=${efivars}/BfCfgBsbSn-${efi_bfcfg_var_guid}
 
   if [ $dump_mode -eq 1 ]; then
-    [ -e "${oob_mac_sysfs}" ] && echo "mfg: MFG_OOB_MAC=$(cat ${oob_mac_sysfs} 2>/dev/null)"
-    [ -e "${opn_sysfs}" ] && echo "mfg: MFG_OPN=$(cat ${opn_sysfs} 2>/dev/null)"
-    [ -e "${sku_sysfs}" ] && echo "mfg: MFG_SKU=$(cat ${sku_sysfs} 2>/dev/null)"
-    [ -e "${modl_sysfs}" ] && echo "mfg: MFG_MODL=$(cat ${modl_sysfs} 2>/dev/null)"
-    [ -e "${sn_sysfs}" ] && echo "mfg: MFG_SN=$(cat ${sn_sysfs} 2>/dev/null)"
-    [ -e "${uuid_sysfs}" ] && echo "mfg: MFG_UUID=$(cat ${uuid_sysfs} 2>/dev/null)"
-    [ -e "${rev_sysfs}" ] && echo "mfg: MFG_REV=$(cat ${rev_sysfs} 2>/dev/null)"
+    [ -e "${oob_mac}" ] && echo "mfg: MFG_OOB_MAC=$(tr -d '\0' < ${oob_mac} 2>/dev/null)"
```

Comment:
Question: this doesn't impact the output when sysfs are used, right?

Reply:
The same line can generate identical output for sysfs and EFI variables:

```
root@bu-lab107-oob:~# echo "mfg: MFG_OOB_MAC=$(tr -d '\0' < /sys/bus/platform/devices/MLNXBF04:00/driver/oob_mac 2>/dev/null)"
mfg: MFG_OOB_MAC=94:6D:AE:15:91:B8
 
root@bu-lab107-oob:~# echo "mfg: MFG_OOB_MAC=$(tr -d '\0' < /sys/firmware/efi/efivars/BfCfgOobMac-487ff588-fb71-4b19-9100-ebe067aa1af0 2>/dev/null)"
mfg: MFG_OOB_MAC=94:6D:AE:15:91:B8
```

Reply:
Got it. I did some comparison with the orignal and updated script for dumping operation. Identical output:

```
root@bu-lab107-oob:~# ./bfcfg -d
icm: LARGE_ICM_SIZE=0x0
mfg: MFG_OOB_MAC=94:6D:AE:15:91:B8
mfg: MFG_OPN=900-9D3B4-00EN-EAB
mfg: MFG_SKU=900-9D3B4-00EN-EAB
mfg: MFG_MODL=D3B4
mfg: MFG_SN=MT2251X00019
mfg: MFG_UUID=02547bbdab80ed118000946dae1591a0
mfg: MFG_REV=A3
mfg: MFG_SYS_MFR=Nvidia
mfg: MFG_SYS_PDN=BF3 SmartNIC Main Card
mfg: MFG_SYS_VER=1.0.0
mfg: MFG_BSB_MFR=Nvidia
mfg: MFG_BSB_PDN=BF3 SmartNIC Main Card
mfg: MFG_BSB_VER=1.0.0
mfg: MFG_BSB_SN=MT2251X00019
sys: ENABLE_SMMU=1
sys: DISABLE_SPMI=0
sys: ENABLE_2ND_EMMC=0
sys: BOOT_PROTECT=0
sys: ENABLE_SPCR=0
sys: DISABLE_PCIE=0
sys: ENABLE_OPTEE=0
sys: ENABLE_I2C0=0
sys: DISABLE_FORCE_PXE_RETRY=0
sys: ENABLE_BMC_FIELD_MODE=0
sys: ENABLE_REDFISH=0
sys: RTCSYNC=0
misc: NET_RSHIM_MAC=00:1a:ca:ff:ff:01
misc: PXE_DHCP_CLASS_ID=NVIDIA/BF/PXE

root@bu-lab107-oob:~# bfcfg -d
icm: LARGE_ICM_SIZE=0x0
mfg: MFG_OOB_MAC=94:6D:AE:15:91:B8
mfg: MFG_OPN=900-9D3B4-00EN-EAB
mfg: MFG_SKU=900-9D3B4-00EN-EAB
mfg: MFG_MODL=D3B4
mfg: MFG_SN=MT2251X00019
mfg: MFG_UUID=02547bbdab80ed118000946dae1591a0
mfg: MFG_REV=A3
mfg: MFG_SYS_MFR=Nvidia
mfg: MFG_SYS_PDN=BF3 SmartNIC Main Card
mfg: MFG_SYS_VER=1.0.0
mfg: MFG_BSB_MFR=Nvidia
mfg: MFG_BSB_PDN=BF3 SmartNIC Main Card
mfg: MFG_BSB_VER=1.0.0
mfg: MFG_BSB_SN=MT2251X00019
sys: ENABLE_SMMU=1
sys: DISABLE_SPMI=0
sys: ENABLE_2ND_EMMC=0
sys: BOOT_PROTECT=0
sys: ENABLE_SPCR=0
sys: DISABLE_PCIE=0
sys: ENABLE_OPTEE=0
sys: ENABLE_I2C0=0
sys: DISABLE_FORCE_PXE_RETRY=0
sys: ENABLE_BMC_FIELD_MODE=0
misc: NET_RSHIM_MAC=00:1a:ca:ff:ff:01
misc: PXE_DHCP_CLASS_ID=NVIDIA/BF/PXE
```